sdk/webpubsub/Microsoft.Azure.WebPubSub.Common/src/WebPubSubConnectionContext.cs

```diff
@@ -87,8 +89,34 @@ public WebPubSubConnectionContext(WebPubSubEventType eventType, string eventName
             UserId = userId;
             Signature = signature;
             Origin = origin;
-            States = states;
-            Headers = headers;
+            States = states ?? new Dictionary<string, object>();
+            Headers = headers ?? new Dictionary<string, string[]>();
+        }
+
+        /// <summary>
+        /// Method to deserialize state to customized type.
+        /// </summary>
+        /// <typeparam name="T">Customized type.</typeparam>
+        /// <param name="stateKey">State key.</param>
+        /// <param name="value">Connection state mapped to the key.</param>
+        /// <returns>Returns true if is able to find the key and false means key doesn't exist.</returns>
+        public bool TryGetState<T>(string stateKey, out T value)
```

Comment:
what happens to `TryGetState<int>` for value `null`?

Reply:
T not supports nullable.